src/bus/webSocketConnection.spec.js

```diff
@@ -442,6 +325,139 @@ describe('Bus/WebSocketConnection', function() {
     });
   });
 
+  describe('onError', function() {
+    let expectedError;
+    let expectedOrganization;
+    let ws;
+
+    beforeEach(function() {
+      expectedError = faker.random.words();
+      expectedOrganization = fixture.build('organization');
+
+      ws = new WebSocketConnection(expectedWebSocket, expectedOrganization.id);
+
+      ws._messageHandlers = {
+        [faker.random.uuid()]: this.sandbox.stub()
+      };
+    });
+
+    it('resets the messageHandlers', function() {
+      try {
+        ws.onError(expectedError);
+      } catch (err) {
+        expect(ws._messageHandlers).to.be.empty;
+      }
+    });
+  });
+
+  describe('onMessage', function() {
+    context('when message handlers exist for the message id', function() {
+      context('when the message is a valid json string', function() {
+        let expectedMessage;
+        let expectedMessageHandlers;
+        let expectedOrganization;
+        let expectedUUID;
+        let ws;
+
+        beforeEach(function() {
+          expectedOrganization = fixture.build('organization');
+          expectedUUID = faker.random.uuid();
+          expectedMessage = {
+            jsonrpc: '2.0',
+            id: expectedUUID,
+            result: null
+          };
+          expectedMessageHandlers = {
+            [expectedUUID]: this.sandbox.stub()
+          };
+
+          ws = new WebSocketConnection(
+            expectedWebSocket,
+            expectedOrganization.id
+          );
+
+          ws._messageHandlers = expectedMessageHandlers;
+
+          ws.onMessage({ data: JSON.stringify(expectedMessage) });
+        });
+
+        it('calls the onmessage function for the message handler', function() {
+          expect(expectedMessageHandlers[expectedUUID]).to.be.calledWith(
+            expectedMessage
+          );
+        });
+      });
+
+      context('when the message is not a valid json string', function() {
+        let expectedMessage;
+        let expectedMessageHandlers;
+        let expectedOrganization;
+        let onMessage;
+        let ws;
+
+        beforeEach(function() {
+          expectedMessage = {
+            jsonrpc: '2.0',
+            id: faker.random.uuid(),
+            result: null
+          };
+          expectedOrganization = fixture.build('organization');
+
+          ws = new WebSocketConnection(
+            expectedWebSocket,
+            expectedOrganization.id
+          );
+
+          ws._messageHandlers = expectedMessageHandlers;
+
+          onMessage = this.sandbox.spy(ws, 'onMessage');
+        });
+
+        it('throws an error', function() {
+          try {
+            ws.onMessage({ data: expectedMessage });
+          } catch (err) {
+            expect(onMessage).to.throw('Invalid JSON in message');
+          }
+        });
```

Comment:
Could you change this to:
```
it('throws an error', function() {
  expect(() => ws.onMessage({ data: expectedMessage })).to.throw(
    'Invalid JSON in message'
  );
});
```

Reasoning behind this? If you were to change your catch to do, say an early return, what is here will not get tested anymore, but the revised version would still show up as a failing test.